src/navigation/reducer.js

```diff
@@ -3,23 +3,15 @@
  * Sustainable Solutions (NZ) Ltd. 2019
  */
 
-import { Navigator } from './Navigator';
+import Navigator from './Navigator';
 
-const paramsEqual = (params1, params2) => {
-  if (params1 === params2) return true;
-
-  if (!params1 || !params2 || Object.keys(params1).length !== Object.keys(params2).length) {
-    return false;
-  }
+const initialState = Navigator.router.getStateForAction(
+  Navigator.router.getActionForPathAndParams('root'),
+);
 
-  return Object.entries(params1).every(([key, value]) => {
-    return value === params2[key];
-  });
-};
-
-export const reducer = (state, action) => {
-  // Prevent the same route being pushed twice (e.g. double tap of button).
-  // See https://github.com/react-community/react-navigation/issues/135.
+const navReducer = (state = initialState, action) => {
```

Comment:
It seems a lot of this code is not longer needed since this react-navigation PR [here](https://github.com/react-navigation/react-navigation/pull/3393)

We can remove the code handling quick double tapping, which includes the `paramsEqual` method?

nit: could make this `const navReducer = (state = initialState, { type }) => { ... }` 
Also I much prefer navigation over nav

Reply:
Resolved with 7011276